src/utils/ignore.js

```diff
@@ -44,8 +44,31 @@ async function createIsIgnoredFunction(ignoreFilePath, withNodeModules) {
 }
 
 /**
- * @param {string} filepath
- * @param {{ignorePath?: string, withNodeModules?: boolean}} options
+ * @param {string[]} ignoreFilePaths
+ * @param {boolean?} withNodeModules
+ * @returns {Promise<(string) => boolean>}
+ */
+async function createIsIgnoredFunction(ignoreFilePaths, withNodeModules) {
+  // If `ignoreFilePaths` is empty, we still want `withNodeModules` to work
+  if (ignoreFilePaths.length === 0 && !withNodeModules) {
+    ignoreFilePaths = [undefined];
+  }
```

Comment:
What if we first read all files in a loop, combine them and then call `createIgnore({ allowRelativePaths: true }).add(combinedContent);`? That's how I was imagining multiple ignore paths to work from the user perspective. This can simplify how `withNodeModules` works: it will be just a new line appended to `combinedContent`.

Example setup where a combined approach kinda matters: [🐸 njt / .prettierignore](https://github.com/kachkaev/njt/blob/main/.prettierignore)

I can imagine removing seconds named `Same as in .gitignore` and `Shared between linters` and add this to `.prettierrc.cjs`: `ignorePath: [".prettierignore", ".gitignore", ".sharedlintignore"]`

Reply:
1, they can be in different directories, they can't be simply concated. We need to know what the patterns related to.

```
#.ignore1
foo

#dir/.ignore2
bar
```

2. In this case

```
#.ignore1
foo/a.js

#.ignore2
!foo/*
foo/b.js
```

should `foo/a.js` be ignored? I think user may expected it be ignored, because it's ignored `.ignore1` file. But if we treat this case like

```
#.ignore
foo/a.js
!foo/*
foo/b.js
```

I'm not sure what's expected.

Reply:
Oh I see... I forgot about ignore files sitting in different directories 🤔 You are right then, we need multiple calls to `createIgnore`

Reply:
@fisker sorry to ask _so late_ ; it is not told in the docs if the multiple ignore paths would respect their own root folder. for example, how would the final ignore list look like for `prettier --ignore-path .ignore1,subfolder/.ignore2 --check subfolder/`? (usecase is: running prettier at root level of a monorepo)